int16 max_time
----
int16 return_code
